source/material_model/steinberger.cc

```diff
@@ -551,7 +580,21 @@ namespace aspect
                                   + " fields of type chemical composition."));
 
           has_background_field = (equation_of_state.number_of_lookups() == n_chemical_fields + 1);
-
+          std::vector<std::string> list_of_composition_names = this->introspection().get_composition_names();
+          if (equation_of_state.number_of_lookups() == 1)
+            list_of_composition_names.resize(1, "background");
+          if ((equation_of_state.number_of_lookups() == 1) && (has_background_field))
+            list_of_composition_names.resize(0, "background");
+          // if (equation_of_state.number_of_lookups() > 1)
+          // list_of_composition_names.resize(equation_of_state.number_of_lookups(), "background");
+          // if ((equation_of_state.number_of_lookups() > 1) && (has_background_field))
+          // list_of_composition_names.resize(equation_of_state.number_of_lookups(), "background");
+          if (equation_of_state.number_of_lookups() > 1)
+            list_of_composition_names.resize(n_chemical_fields, "background");
```

Comment:
The line that you've added here appends to the list_of_composition_names vector with entries called "background" (e.g. {"a", "b", "background", "background", "background", "background", ...}.

I doubt that is what you want to do. The background field is the first compositional field, not the last. And there is only ever one of them. 

Have another look at the link above. I suspect "insert" will be more useful to you than "resize".

Reply:
If you're ever unsure of what a C++ function does, you can always look it up: https://cplusplus.com/reference/vector/vector/resize/
https://cplusplus.com/reference/vector/vector/insert/